src/goto-programs/assign_params_as_non_det.cpp

```diff
@@ -36,24 +57,200 @@ bool assign_params_as_non_det::runOnFunction(
       return false; // Not expected
     exprt lhs = symbol_expr(*context.find_symbol(_id));
 
-    // rhs
-    exprt rhs = exprt("sideeffect", lhs.type());
-    rhs.statement("nondet");
-    rhs.location() = l;
+    if (lhs.type().is_pointer())
+    {
+      // e.g. int* lhs;
+      // to
+      // lhs = null;
+      // bool temp;
+      // temp = nondet();
+      // if(temp)
+      // {
+      //   int temp2;
+      //   temp2 = nondet();
+      //   assume(temp2 != 0);
+      //   lhs = &temp2;
+      // }
+
```

Comment:
we need to add a type check here. If it's a void pointer we cannot dereference it

Reply:
Yes, How to cover this, just put in a warning? Saying that "Pointer details cannot be accessed because a void pointer is in the params, please change method to not have void* if you want to see their contents" maybe?